src/Components/Assets/configure/MonitorConfigure.tsx

```diff
@@ -1,52 +1,51 @@
-import { Dispatch, useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 import { BedSelect } from "../../Common/BedSelect";
 import { BedModel } from "../../Facility/models";
 import { AssetData } from "../AssetTypes";
-import {
-  createAssetBed,
-  listAssetBeds,
-  partialUpdateAssetBed,
-} from "../../../Redux/actions";
 import * as Notification from "../../../Utils/Notifications.js";
-import { useDispatch } from "react-redux";
 import { Submit } from "../../Common/components/ButtonV2";
 import { FieldLabel } from "../../Form/FormFields/FormField";
+import request from "../../../Utils/request/request";
+import routes from "../../../Redux/api";
 
-const saveLink = (assetId: string, bedId: string, dispatch: Dispatch<any>) => {
-  dispatch(createAssetBed({}, assetId, bedId));
+const saveLink = async (assetId: string, bedId: string) => {
+  await request(routes.createAssetBed, {
+    body: {
+      asset: assetId,
+      bed: bedId,
+    },
+  });
   Notification.Success({ msg: "AssetBed Link created successfully" });
 };
-const update_Link = (
+const update_Link = async (
   assetbedId: string,
   assetId: string,
-  bed: BedModel,
-  assetBed: any,
-  dispatch: Dispatch<any>
+  bed: BedModel
 ) => {
-  dispatch(
-    partialUpdateAssetBed(
-      {
-        asset: assetId,
-        bed: bed.id,
-      },
-      assetbedId
-    )
-  );
+  await request(routes.partialUpdateAssetBed, {
+    pathParams: { external_id: assetbedId },
+    body: {
+      asset: assetId,
+      bed: bed.id ?? "",
+    },
+  });
   Notification.Success({ msg: "AssetBed Link updated successfully" });
 };
 
 export default function MonitorConfigure({ asset }: { asset: AssetData }) {
   const [bed, setBed] = useState<BedModel>({});
   const [updateLink, setUpdateLink] = useState<boolean>(false);
   const [assetBed, setAssetBed] = useState<any>();
-  const dispatch: any = useDispatch();
 
   const getAssetBeds = async (id: string) => {
-    const assetBeds = await dispatch(listAssetBeds({ asset: id }));
-    if (assetBeds.data?.results?.length > 0) {
+    // const assetBeds = await dispatch(listAssetBeds({ asset: id }));
+    const { data } = await request(routes.listAssetBeds, {
+      query: { asset: id },
+    });
+    if (data && data.results?.length > 0) {
       setUpdateLink(true);
-      setAssetBed(assetBeds.data.results[0]);
-      setBed(assetBeds.data.results[0].bed_object);
+      setAssetBed(data.results[0]);
+      setBed(data.results[0].bed_object);
```

Comment:
These states are also not needed right?

`useQuery` is designed such that we no longer need to maintain separate state variables. `useQuery` already does perform this internally by having `[data, setData] = useState<T>()` for us.

Reply:
done

Reply:
setBed is being passed to BedSelect component as callback so i left t as it is